test/unit/pages/selection/controller-test.js

```diff
@@ -304,45 +304,35 @@ describe('SelectionController', function() {
     });
   });
 
-  describe('download', function() {
-    it('should open url in browser using electron.shell.openExternal', function() {
-      sandbox.stub(electron.shell);
-      selectionController.download('url');
-      expect(electron.shell.openExternal).calledOnce;
-    });
-  });
-
   describe('detectInstalledComponents', function() {
     it('should return existing promise if detection is already running', function() {
       let detection = selectionController.detectInstalledComponents();
       expect(detection).equals(selectionController.detectInstalledComponents());
     });
   });
 
-  describe('selectAll', function() {
+  describe('toggleSelectAll', function() {
     beforeEach(inject(context));
     it('should not select detected components', function() {
       let kompose = selectionController.sc.checkboxModel.kompose;
       kompose.addOption('detected', '1.0.0', 'location', true);
       kompose.selectedOption = 'detected';
-      selectionController.selectAll();
+      selectionController.toggleSelectAll();
       expect(kompose.selectedOption).equals('detected');
     });
 
     it('should select all installable components', function() {
       let kompose = selectionController.sc.checkboxModel.kompose;
-      selectionController.selectAll();
+      selectionController.toggleSelectAll();
       expect(kompose.selectedOption).equals('install');
     });
-  });
 
-  describe('deselectAll', function() {
-    beforeEach(inject(context));
-    it('should deselect all detected components', function() {
-      selectionController.deselectAll();
+    it('should deselect all selected components', function() {
+      selectionController.isAllSelected = false;
+      selectionController.toggleSelectAll();
       for (let installer of installerDataSvc.allInstallables().values()) {
```

Comment:
you should iterate over controller.checkboxModel, rater through all installables in installerDataSvc.
``` javascript
  for (let installer in selectionController.checkboxModel) {
    expect(installer.selectedOption).equals('detected');
  }
```

Reply:
Fixed this.